pilot/cmd/pilot-agent/main.go

```diff
@@ -178,11 +178,12 @@ var (
 			// resolve statsd address
 			if proxyConfig.StatsdUdpAddress != "" {
 				addr, err := proxy.ResolveAddr(proxyConfig.StatsdUdpAddress)
-				if err == nil {
-					proxyConfig.StatsdUdpAddress = addr
+				if err != nil {
+					// If istio-mixer.istio-system can't be resolved, skip generating the statsd config.
+					// (instead of crashing). Mixer is optional.
+					log.Warnf("resolve StatsdUdpAddress failed: %v", err)
 				}
-				// If istio-mixer.istio-system can't be resolved, skip generating the statsd config.
-				// (instead of crashing). Mixer is optional.
+				proxyConfig.StatsdUdpAddress = addr
```

Comment:
you need an else here. And unset the statsdudp address if its not resolvable. It will help a lot of failed deployments

Reply:
SGTM

Reply:
addressed